python/mp/rosary.py

```diff
@@ -189,12 +245,14 @@ def add_effect(self, name, bead_set_name='all', color_name_or_r=None,
                 effect_color = color.Color(color_name_or_r, g, b)
 
         # If all else fails, just pick a random color from the registry
-        if effect_color is None:
+        while effect_color in (None, color.Color(0,0,0)):
             effect_color = random.choice(list(self.color_registry.values()))
+
         print("USING COLOR: {}".format(effect_color))
 
         return self.add_effect_object(self.effect_registry[name](bead_set,
-                                                                 effect_color))
+                                                                 effect_color,
+                                                                 duration))
```

Comment:
Let's try to do things with triggers and effects before designing too much in here. We can add more abstraction after we see where the friction is in the API.